### PR TITLE
Variables: replaces homegrown variableAdapters with Registry

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
When we created the newVariables branch for variables that were migrating to React/Redux, we created variableAdapters and the VariableAdapter interface that is used in call/create/query of variable types. One comment in that PR was that maybe we should use the `Registry` class instead of using variableAdapters. This PR tries to replace variableAdapters with Registry. 

Not sure if I use it as intended and if I could use more of the Registry in this domain.

**Which issue(s) this PR fixes**:
Relates to #22564

**Special notes for your reviewer**:
Would love to get feedback if this was what you were expecting @dprokop @torkelo

### CODE REVIEW

File: public/app/features/variables/constant/actions.test.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ import { setCurrentVariableValue } from '../state/sharedReducer';
 import { initDashboardTemplating } from '../state/actions';
 
 describe('constant actions', () => {
-  variableAdapters.set('constant', createConstantVariableAdapter());
+  variableAdapters.setInit(() => [createConstantVariableAdapter()]);
```
Comment (torkelo) [inline public/app/features/variables/constant/actions.test.ts]: really like that all the definitions are now lazy :)

**Review by torkelo — APPROVED** [review]
This looks really good!

File: public/app/features/variables/constant/actions.test.ts (hunk 1 above)
Comment (hugohaggmark) [reply public/app/features/variables/constant/actions.test.ts]: Clicked the confused reaction instead of thumbs up, sorry for that

**Review by mckn — APPROVED** [review]
LGTM

**Review by dprokop — APPROVED** [review]
Great work @hugohaggmark, as always :)

**Comment by hugohaggmark** [comment]
> Great work @hugohaggmark, as always :)

Thank you so much @dprokop !

**Pull request merged**